pkg/apis/apps/validation/validation_test.go

```diff
@@ -62,53 +62,104 @@ func makeSet(selectorLabels *map[string]string, template *api.PodTemplate, tweak
 	return s
 }
 
-func tweakName(name string) setTweak {
-	return func(s apps.StatefulSet) {
-		s.ObjectMeta.Name = name
+func tweakName(name string) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.ObjectMeta.Name = name
 	}
 }
 
-func tweakNamespace(ns string) setTweak {
-	return func(s apps.StatefulSet) {
-		s.ObjectMeta.Namespace = ns
+func tweakNamespace(ns string) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.ObjectMeta.Namespace = ns
 	}
 }
 
-func tweakManagementPolicy(policy apps.PodManagementPolicyType) setTweak {
-	return func(s apps.StatefulSet) {
-		s.Spec.PodManagementPolicy = policy
+func tweakMetaLabels(labels map[string]string) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.ObjectMeta.Labels = labels
 	}
 }
 
-func tweakRepicas(replicas int32) setTweak {
-	return func(s apps.StatefulSet) {
-		s.Spec.Replicas = replicas
+func tweakAnnotations(annotations map[string]string) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.ObjectMeta.Annotations = annotations
 	}
 }
 
-func tweakPodManagementPolicy(policy apps.PodManagementPolicyType) setTweak {
-	return func(s apps.StatefulSet) {
-		s.Spec.PodManagementPolicy = policy
+func tweakFinalizers(finalizers []string) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.ObjectMeta.Finalizers = finalizers
 	}
 }
 
-func tweakUpdateStrategy(strategy apps.StatefulSetUpdateStrategy) setTweak {
-	return func(s apps.StatefulSet) {
-		s.Spec.UpdateStrategy = strategy
+func tweakManagementPolicy(policy apps.PodManagementPolicyType) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.Spec.PodManagementPolicy = policy
 	}
 }
 
-func tweakMinReadySeconds(t int32) setTweak {
-	return func(s apps.StatefulSet) {
-		s.Spec.MinReadySeconds = t
+func tweakReplicas(replicas int32) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.Spec.Replicas = replicas
+	}
+}
+
+func tweakPodManagementPolicy(policy apps.PodManagementPolicyType) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.Spec.PodManagementPolicy = policy
+	}
+}
+
+func tweakSelectorLabels(labels map[string]string) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.Spec.Selector = &metav1.LabelSelector{MatchLabels: labels}
+	}
+}
+
+func tweakRestartPolicy(rp api.RestartPolicy) statefulSetTweak {
```

Comment:
I'd include the word Template in the name here

Reply:
modified as suggested